core/instance/Instance.js

```diff
@@ -320,14 +320,16 @@ class Instance extends EventEmitter {
         const children = this.parents.get(id)
 
         if (children && children.size > 0) {
-            console.log(id, 'had children', children)
-            throw new Error('Cannot remove entity without removing its components first.')
+            children.forEach(nid => {
+                const component = { [this.config.ID_PROPERTY_NAME]: nid }
+                this.removeComponent(component, entity)
```

Comment:
@timetocode does it actually remove the component if a separate object is allocated?